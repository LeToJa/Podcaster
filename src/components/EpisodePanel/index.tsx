import { Parser } from 'html-to-react'

import AudioPlayer from '../../components/AudioPlayer'

import { RSSResponse } from '../../helpers/RSSParser/types'

const EpisodePanel = ({ title, content, enclosure: { url, type } }: RSSResponse) => <>
    <div className='shadow-xl p-4'>
        <blockquote className='text-sm'>
            <h2 className='font-bold text-3xl mb-4'>{title}</h2>
            {Parser().parse(content)}
        </blockquote>
        <figure className='flex justify-center mt-6'>
            <AudioPlayer url={url} type={type} />
        </figure>
    </div>
</>

export default EpisodePanel