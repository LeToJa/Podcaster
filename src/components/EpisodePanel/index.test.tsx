import { render } from '@testing-library/react'
import { describe, it, expect, vi } from 'vitest'

import EpisodePanel from '.'

vi.mock('../../components/AudioPlayer', () => ({
    default: vi.fn(() => <figure>Audio Player Mock</figure>),
}))

vi.mock('html-to-react', () => ({
    Parser: () => ({
        parse: vi.fn((htmlContent: string) => <>{htmlContent}</>),
    }),
}))

const mockEpisode = {
    title: 'Test Episode',
    content: '<p>Episode content in HTML format</p>',
    isoDate: '2023-08-29T12:00:00Z',
    enclosure: {
        url: 'http://example.com/audio.mp3',
        type: 'audio/mpeg',
    },
    itunes: {
        duration: '30:00',
    }
}

describe('EpisodePanel', () => {
    it('renders correctly and matches snapshot', () => {
        const { container } = render(<EpisodePanel {...mockEpisode} />)

        expect(container).toMatchSnapshot()
    })

    it('renders correctly with given episode data', () => {
        const { getByText } = render(<EpisodePanel {...mockEpisode} />)

        expect(getByText(/Test Episode/i)).toBeTruthy()
        expect(getByText(/Episode content in HTML format/i)).toBeTruthy()
        expect(getByText(/Audio Player Mock/i)).toBeTruthy()
    })
})
