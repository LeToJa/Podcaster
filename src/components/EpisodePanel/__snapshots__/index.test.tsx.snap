// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EpisodePanel > renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="shadow-xl p-4"
  >
    <blockquote
      class="text-sm"
    >
      <h2
        class="font-bold text-3xl mb-4"
      >
        Test Episode
      </h2>
      &lt;p&gt;Episode content in HTML format&lt;/p&gt;
    </blockquote>
    <figure
      class="flex justify-center mt-6"
    >
      <figure>
        Audio Player Mock
      </figure>
    </figure>
  </div>
</div>
`;
